src/test/java/streams/part2/exercise/Exercise1.java

```diff
@@ -55,9 +69,14 @@ public void composeFullNamesOfEmployeesUsingLineSeparatorAsDelimiter() {
     @SuppressWarnings("Duplicates")
     public void groupPersonsByFirstPositionUsingToMap() {
         List<Employee> employees = getEmployees();
-
-        // TODO реализация
-        Map<String, Set<Person>> result = null;
+        Function<Employee, Set<Person>> f = employee -> new HashSet<>(
+            Collections.singletonList(employee.getPerson()));
+        Map<String, Set<Person>> result = employees.stream()
+            .collect(Collectors.toMap(employee -> employee.getJobHistory().get(0).getPosition(), employee -> new HashSet<>(
```

Comment:
Why do you send new HashSet as function. I just don't understand. It seems to me, we have to send function, which returns Person here (generic parameter U in toMap). toMap will return Collector with Map<K, U>(Map<String, Person> in our case). In my mind, collector with Map<String, HashMap<Person>> is returned here, following your code.